packages/domain-picker/src/domain-picker/index.tsx

```diff
@@ -199,6 +199,17 @@ const DomainPicker: FunctionComponent< Props > = ( {
 		}
 	}, [ initialDomainSearch, showSearchField ] );
 
+	useEffect( () => {
+		// when people unfold for more results, focus on the first item after the fold
+		// this is not the cleanest solution, but still cleaner than useRef spaghetti
+		if ( isExpanded ) {
+			const focusableElements = document.querySelectorAll(
+				'.domain-picker__suggestion-select-button, button.domain-picker__suggestion-item'
```

Comment:
These two classes are used in a different component and with the conditionals and wrapping happening in SuggestionItem, it's not straightforward to understand or to debug the use cases.

Using ref would be a clean solution for this parent component and it will also enable future imperative manipulation such as "scroll to item on mobile" if needed.

I've opened https://github.com/Automattic/wp-calypso/pull/48631 with a proposal in this direction.

Reply:
Your implementation is much cleaner than I thought the refs way would be. Thanks!

I'll review it.